Letterbook.Api/Controllers/ActivityPub/ActorController.cs

```diff
@@ -27,6 +28,7 @@ namespace Letterbook.Api.Controllers.ActivityPub;
 [Consumes("application/ld+json",
 	"application/ld+json; profile=\"https://www.w3.org/ns/activitystreams\"",
 	"application/activity+json")]
+[Authorize(policy: "ActivityPub")]
```

Comment:
I had to do some refresher reading, but there's multiple ways to vary the policy depending on the specific requirements and your preferred design.
- An obvious but clumsy alternative: multiple policies
- An imperative authentication alternative: explicitly invoke authentication with the Signature scheme and act based on the result, dropping the AuthorizeAttribute
- An authz-policy-based alternative: Custom-attribute-driven policy metadata per endpoint
- Imperative authz-policy-based alternative

For producing different outputs, purely declarative authorization would probably become unnecessarily convoluted. That would possibly be better served with imperative authorization checks, where the handlers can even be given access to an object representing the resource being accessed, e.g. some AP object.

If you want, say, a controller where there's one or two exceptions that should negotiate responses instead of outright forbidding the request, you could slap an `[AllowAnonymous]` on the methods, do an explicit check there and let the default work for everything else.

Another thing also occurred to me: should we ever actually allow a signed request through, if the verification fails? Right now, that results in no principal and the authz policy will deny the requests _when the policy is evaluated_, but it would allow to invoke any non-authorizing endpoints and the invalid signature would be ignored.

Reply:
Um, good question. My first instinct is to say yes, we should allow (limited) anonymous access when verification fails. But, I think that puts a lot of faith in peer servers to actually refresh their caches and (and thus try again later, hopefully with fixed signatures). And I'm not sure that faith is warranted. So it might be better to fail loudly than to partially succeed.

Reply:
I think I'll rework this a bit before merging, then:

- Put the signature validation in the middleware pipeline
- Always validate if signatures are present
- Fail requests containing invalid signatures with a 400
- Stick the successful result in a request feature or equivalent
- Have the handler query the result from there

This shouldn't be too difficult, but I think it will result in a pretty neat separation of concerns.